metricbeat/module/nifi/flowfilerepostorage/data.go

```diff
@@ -0,0 +1,55 @@
+package flowfilerepostorage
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// FlowFileRepositoryStorageUsage ...
+type FlowFileRepositoryStorageUsage struct {
+	FreeSpace       string `json:"freeSpace"`
+	FreeSpaceBytes  uint64 `json:"freeSpaceBytes"`
+	TotalSpace      string `json:"totalSpace"`
+	TotalSpaceBytes uint64 `json:"totalSpaceBytes"`
+	UsedSpace       string `json:"usedSpace"`
+	UsedSpaceBytes  uint64 `json:"usedSpaceBytes"`
+	Utilization     string `json:"utilization"`
+}
+
+// Data ...
+type Data struct {
+	SystemDiagnostics struct {
+		AggregateSnapshot struct {
+			FlowFileRepositoryStorageUsage `json:"flowFileRepositoryStorageUsage"`
+		} `json:"aggregateSnapshot"`
+	} `json:"systemDiagnostics"`
+}
+
+func eventMapping(body io.Reader) common.MapStr {
+	var data Data
+	err := json.NewDecoder(body).Decode(&data)
+	if err != nil {
+		logp.Err("Error: ", err)
+	}
+
+	slice := data.SystemDiagnostics.AggregateSnapshot.FlowFileRepositoryStorageUsage
+
+	fmt.Printf("%v", data)
+	fmt.Printf("%v", slice)
+
+	event := common.MapStr{
```

Comment:
@ruflin 

I'd certainly prefer to use the `schema` magic here, but haven't figured out how yet. I need to do a bit of slicing/dicing of NiFi's API response to avoid array types, and break the response into different metricsets. 

I've uploaded [sample output](https://github.com/scottcrespo/beats/blob/metricbeat-nifi-module/metricbeat/module/nifi/flowfilerepostorage/_meta/sample_output.json) so you can see what I'm trying to work with. Note there are fields titled `aggregateSnapshot` and `nodeSnapshots`. I'm currently working on an implementation that interrogates a target to figure out if it's a member of a cluster or standalone, and use `aggregateSnapshot` or `nodeSnapshots` accordingly. 

I've based my implementation off of the couchbase module's [data.go](https://github.com/elastic/beats/blob/master/metricbeat/module/couchbase/bucket/data.go) which appears to deal with the same array type problem using structs and `common.MapStr`. Is this a suitable model to follow?

So to use the `schema` package here... I think I can distill my question down to this: How can I create a `schema` instance that either (a) represents a segment of NiFi's API response or (b) can be further processed into the desired event schema with ES-friendly data types?